### DISCUSSION TITLE
ParBilinearForm w/ non-ConstantCoefficients in MassIntegrator taking extremely long to Assemble

### DISCUSSION THREAD

**Discussion by j-signorelli** [mfem/mfem#3740] [Q&A]
Hi,

For H1-conforming elements, I have code that uses two ParBilinearForms: one with just a MassIntegrator and a separate one with just a DiffusionIntegrator. Whenever I use non-ConstantCoefficients (Like GridFunctionCoefficient), the one with the DiffusionIntegrator Assembles quite quickly regardless with no issues, but the one with the MassIntegrator takes an exceptionally long time. When I run with 1 process, it is not too bad, but the more I run with the seemingly expontentially longer it takes. Is this a common problem? Is there anything I can do to speed this up?

Thank you for any assistance. I can provide more information if helpful.

**Comment by j-signorelli** [comment]
Just to add more info to this: Right before submitting this discussion post, I started a run with 14 procs, and over 4 hours later it still is assembling, even though it only took a couple minutes on 1 proc.

**Reply by jandrej** [reply]
Can you post a minimal failing example that we can look to possibly reproduce the issue?

**Reply by j-signorelli** [reply]
Creating a minimal failing example helped me find my issue, thanks! It was nothing MFEM-related.
